pkg/controllers/pod_controller_test.go

```diff
@@ -0,0 +1,687 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
+
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+	"sigs.k8s.io/jobset/pkg/constants"
+	testutils "sigs.k8s.io/jobset/pkg/util/testing"
+)
+
+func TestListPodsForJob(t *testing.T) {
```

Comment:
We can remove this test.

The only thing the method [listPodsForJob](https://github.com/kubernetes-sigs/jobset/blob/8860f832dee2c35d9cb2951260208fcef87cd2db/pkg/controllers/pod_controller.go#L161) does is make a List() API call for pod objects with a specific annotation. 

So we can't really inject an fake List() function via interceptorFunc here for testing, because there's no logic to test, so we would just be testing the test code.

Reply:
So, does that mean I should remove the injected interceptorFunc method?

Reply:
If the interceptorFunc method is not injected, it will not be possible to test the scenario where the listing fails.

Reply:
@danielvegamyhre

Reply:
We should remove this test entirely.

Reply:
done